xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/macros.vm

```diff
@@ -768,16 +768,21 @@ $fn
 #macro(panelheader $title)
 ## Note: We pass the Panel name as an HTML class attribute so that it's possible to style
 ## the Panels selectively using CSS.
-#set ($specialClassAttribute = '')
-#if ($paneldoc)
-  #set ($specialClassAttribute = $util.convertToAlphaNumeric($paneldoc.name))
+#set ($defaultDocumentName = $services.model.getEntityReference('DOCUMENT', 'default').name)
+#if ($paneldoc.documentReference.name == $defaultDocumentName)
```

Comment:
Missing some doc here to explain why we do that. 

For example: "We want to avoid showing the "WebHome" suffix in the generated class since we're working to remove it from the UI and from the APIs. In addition it generates class names longer than they are needed. Thus for nested pages we strip it".

Reply:
Maybe also explain we don't want to display the wiki prefix to have a shorter generated class name